hack/build_speed.sh

```diff
@@ -0,0 +1,126 @@
+#! /bin/sh
+
+# The main goal of this script is test and time builds using Buildah or Docker.
+# We hope to use it to help optimize Buildah build performance
+#
+# It takes two options
+# First option tells the type of the container image
+# build to do.  Valid options are:
+#       Docker  - docker build
+#       Buildah - buildah bud
+#       Both    - Do docker build followed by buildah bud
+#
+# Second Option specifies a directory or cleanup
+# The script will 'find' files beginning with Dockerfile, for each Dockerfile
+# it finds it will run a build with the Dockerfile and directory for the
+# context. When it does the builds, it will call time on them to show how
+# long the builds take.  The created image name will be a combination of the
+# lowercased Directory name that the Dockerfile was found in plus the lower
+# cased dockerfile name.
+#
+# if the second field is cleanup, the script will remove all images from the
+# specified builder.
+#
+# The script does not check for conflicts on nameing.
+#
+# Outputs file:
+#
+#
+# cat /tmp/buildspeedllN.json
+#  [{"engine": "docker", "command": "docker build --no-cache -f /root/Dockerfile.test -t root_dockerfile.test /root", "real": "1:36.23"},
+#  {"engine": "buildah", "command": "buildah bud --layers -f /root/Dockerfile.test -t root_dockerfile.test /root", "real": "0:01.10"}]
+#
+# Examples uses
+#     ./build_speed.sh Docker ~/MyImages
+#     ./build_speed.sh Both /usr/share/fedora-dockerfiles/django/Dockerfile
+
+#totalsfile=$(mktemp /tmp/buildspeedXXX.json)
+totalsfile=/tmp/build_speed.json
+echo -n '{'>$totalsfile
+firstBoth=1
+
+Dockerfiles() {
+    find -L $1 -name Dockerfile\*
+}
+
+Buildah() {
+    Name=$1
+    Dockerfile=$2
+    Context=$3
+    echo buildah bud --layers -f ${Dockerfile} -t ${Name} ${Context}
+    Time buildah bud --layers -f ${Dockerfile} -t ${Name} ${Context}
+}
+
+Time() {
+    outfile=$(mktemp /tmp/buildspeedXXX)
+    /usr/bin/time -o $outfile --f "%E" $@
+    echo "{\"engine\": \"$1\", \"command\": \"$@\", \"real\": \"$(cat ${outfile})\"}"
```

Comment:
hint: `$(< outfile)` (dollar paren less-than) is equivalent to `$(cat` and, some believe, shorter and more readable.